crates/parser/src/ix_parser/token_program/token_ix_context.rs

```diff
@@ -0,0 +1,148 @@
+    pub freeze_authority: Option<Pubkey>,
+}
+#[derive(Debug)]
+pub struct InitializeAccount {
+    pub account: Pubkey,
+    pub mint: Pubkey,
+    pub owner: Pubkey,
+}
+#[derive(Debug)]
+pub struct InitializeMultisig {
+    pub multisig: Pubkey,
+    pub signers: Vec<Pubkey>,
+    pub m: u8,
+}
+#[derive(Debug)]
+pub struct Approve {
+    pub source: Pubkey,
+    pub delegate: Pubkey,
+    pub amount: u64,
+}
+#[derive(Debug)]
+pub struct Revoke {
+    pub source: Pubkey,
+}
+#[derive(Debug)]
+pub struct SetAuthority {
+    pub owned: Pubkey,
+    pub authority_type: AuthorityType,
+    pub new_authority: COption<Pubkey>,
+}
+#[derive(Debug)]
+pub struct MintTo {
+    pub mint: Pubkey,
+    pub account: Pubkey,
+    pub amount: u64,
+}
+#[derive(Debug)]
+pub struct Burn {
+    pub account: Pubkey,
+    pub mint: Pubkey,
+    pub amount: u64,
+}
+#[derive(Debug)]
+pub struct CloseAccount {
+    pub account: Pubkey,
+    pub destination: Pubkey,
+}
+#[derive(Debug)]
+pub struct FreezeAccount {
+    pub account: Pubkey,
+    pub mint: Pubkey,
+}
+#[derive(Debug)]
+pub struct ThawAccount {
+    pub account: Pubkey,
+    pub mint: Pubkey,
+}
+#[derive(Debug)]
+pub struct TransferChecked {
+    pub source: Pubkey,
+    pub mint: Pubkey,
+    pub destination: Pubkey,
+    pub amount: u64,
+    pub decimals: u8,
+}
+#[derive(Debug)]
+pub struct ApproveChecked {
+    pub source: Pubkey,
+    pub mint: Pubkey,
+    pub delegate: Pubkey,
+    pub amount: u64,
+    pub decimals: u8,
+}
+#[derive(Debug)]
+pub struct MintToChecked {
+    pub mint: Pubkey,
+    pub account: Pubkey,
+    pub amount: u64,
+    pub decimals: u8,
+}
+#[derive(Debug)]
+pub struct BurnChecked {
+    pub account: Pubkey,
+    pub mint: Pubkey,
+    pub amount: u64,
+    pub decimals: u8,
+}
+#[derive(Debug)]
+pub struct SyncNative {
+    pub account: Pubkey,
+}
+#[derive(Debug)]
+pub struct GetAccountDataSize {
+    pub mint: Pubkey,
+}
+#[derive(Debug)]
+pub struct InitializeImmutableOwner {
+    pub account: Pubkey,
+}
+#[derive(Debug)]
+pub struct AmountToUiAmount {
+    pub mint: Pubkey,
+    pub amount: u64,
+}
+#[derive(Debug)]
+pub struct UiAmountToAmount {
+    pub mint: Pubkey,
+    pub ui_amount: String,
+}
+
+#[derive(Debug)]
+pub enum TokenIxContext {
+    Transfer(Transfer),
+    InitializeMint(InitializeMint),
+    InitializeAccount(InitializeAccount),
+    InitializeMultisig(InitializeMultisig),
+    Approve(Approve),
+    Revoke(Revoke),
+    SetAuthority(SetAuthority),
+    MintTo(MintTo),
+    Burn(Burn),
+    CloseAccount(CloseAccount),
+    FreezeAccount(FreezeAccount),
+    ThawAccount(ThawAccount),
+    TransferChecked(TransferChecked),
+    ApproveChecked(ApproveChecked),
+    MintToChecked(MintToChecked),
+    BurnChecked(BurnChecked),
+    SyncNative(SyncNative),
+    GetAccountDataSize(GetAccountDataSize),
+    InitializeImmutableOwner(InitializeImmutableOwner),
+    AmountToUiAmount(AmountToUiAmount),
+    UiAmountToAmount(UiAmountToAmount),
```

Comment:
What are these instructions and why not use the structs provided by the program?

Reply:
these ixs are from spl token program itself